JEDI-based ensemble recentering and analysis calculation and native grid increments

DavidNew-NOAA opened the pull request:
This PR is a companion to GDASApp PR [#1488](https://github.com/NOAA-EMC/GDASApp/pull/1488) and Global Workflow PR [#3312](https://github.com/NOAA-EMC/global-workflow/pull/3312).

It creates YAMLs for the JEDI-based ensemble recentering application and the `convertstate` applications used for analysis calculation. It also alters certain YAMLs to implement native-grid increments in the Global Workflow.

Some miscellaneous changes to the naming convention of increments and backgrounds in run directories are also implemented for overall consistency and simplicity across JEDI-based Global Workflow jobs.

DavidNew-NOAA committed:
Initial commit

DavidNew-NOAA committed:
Save progress

DavidNew-NOAA committed:
Save progress

DavidNew-NOAA committed:
Save progress

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Update staging and ecen yaml

DavidNew-NOAA committed:
Add cubed_sphere_grid_ prefix back on in data dirs

DavidNew-NOAA committed:
Missed a file

DavidNew-NOAA committed:
Debug

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Save progress

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Update

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Save progress before redoing some stuff

DavidNew-NOAA committed:
Revert some stuff

DavidNew-NOAA committed:
Add convert state algo for analcalc

DavidNew-NOAA committed:
Add algorithm templates for snow and aero analcalc convertstate

DavidNew-NOAA committed:
Revert unnecessary science change (increment resolution)

DavidNew-NOAA committed:
Revert "Revert unnecessary science change (increment resolution)"

This reverts commit 71ef27ddd9c55b2a2deead2bbfbf191492b90724.

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Some renaming

DavidNew-NOAA committed:
Revert a few things

CoryMartin-NOAA commented on algorithm/atmosphere/atm_analcalc_convertstate.yaml.j2:
```diff
@@ -0,0 +1,38 @@
+states:
+{% for fh in iau_hours %}
+- input:
+    datetime: {{ atmosphere_background_time_iso }}
```
won't the background time vary based on the IAU hour?

DavidNew-NOAA replied:
Good catch

DavidNew-NOAA committed:
Address reviewer comments

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Use fv3jedi_add_increments rather than convertstate

DavidNew-NOAA committed:
Restore prefix to history files in rundirs

DavidNew-NOAA committed:
Missed some changes in last commit

DavidNew-NOAA committed:
More missed stuff

DavidNew-NOAA committed:
Create snow and aero variational history prefixes

DavidNew-NOAA committed:
More revisions

DavidNew-NOAA committed:
update client integration test

DavidNew-NOAA committed:
Debug

danholdaway committed:
Add test of ensemble mode

DavidNew-NOAA committed:
Update

DavidNew-NOAA committed:
Merge branch 'feature/create_ensemble_mode' into feature/calcanl

DavidNew-NOAA committed:
Revert "Merge branch 'feature/create_ensemble_mode' into feature/calcanl"

This reverts commit 956165bc7b83f53ceaeed645231f831b3fb17d52, reversing
changes made to cdb6e00120d8fdefbde1f0abbe40130108a0d7c5.

DavidNew-NOAA committed:
Update

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
Update

DavidNew-NOAA committed:
Switich to conventional JEDI addincrement application

DavidNew-NOAA committed:
Add missing template

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

DavidNew-NOAA committed:
wrong fields metadata

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

jiaruidong2017 committed:
Set up the reject list in observation chronicle for GTS synoptic snow depth data (#79)

This PR takes the branch `feature/snow-example-chronicle` from
@CoryMartin-NOAA and adds a initial list of the rejected station IDs in
observation chronicle for GTS synoptic snow depth data by using
https://github.com/NOAA-EMC/jcb/issues/17
This PR also adds/removes additional stations chronologically in the
observation chronicle.

This PR needs JCB updated to the commit hash `b114fe43` or later.

DavidNew-NOAA commented:
@danholdaway @CoryMartin-NOAA @RussTreadon-NOAA Since Global Workflow PR [#3312](https://github.com/NOAA-EMC/global-workflow/pull/3312) is close to final testing, this PR is ready for final review and merger.

DavidNew-NOAA committed:
Merge branch 'develop' into feature/calcanl

CoryMartin-NOAA committed:
Merge branch 'develop' into feature/calcanl

CoryMartin-NOAA commented on algorithm/aero/aero_addincrement.yaml.j2:
```diff
@@ -0,0 +1,43 @@
+  npz: {{ aero_npz_ges }}
+  field metadata override: ./fv3jedi/fv3jedi_fieldmetadata_restart.yaml
+state:
+  datetime: {{ atmosphere_background_time_iso }}
```
aerosol or atmosphere?

CoryMartin-NOAA commented on algorithm/atmosphere/fv3jedi_ensemble_recenter.yaml.j2:
```diff
@@ -0,0 +1,47 @@
+    filetype: cube sphere history
+    provider: ufs
+    datapath: ./
+    filename: enkf{{ atmosphere_ensemble_analysis_prefix }}cubed_sphere_grid_catmi{{ '%03d' % atmosphere_iau_hours[ihour] }}.nc
```
why enkf in front? shouldn't enkf be in the prefix?

CoryMartin-NOAA commented on algorithm/snow/snow_addincrement.yaml.j2:
```diff
@@ -0,0 +1,43 @@
+  npz: {{ snow_npz_ges }}
+  field metadata override: ./fv3jedi/fv3jedi_fieldmetadata_restart.yaml
+state:
+  datetime: {{ atmosphere_background_time_iso }}
```
is it atmosphere or snow?

DavidNew-NOAA committed:
Address reviewer comments

DavidNew-NOAA committed:
Update

DavidNew-NOAA commented:
@CoryMartin-NOAA All changes made and retested
